### PR TITLE
Add lt-qcom project with kernel cases

### PR DESCRIPTION
**Pull request by alimon**

### CODE REVIEW

**Comment by roxell** [comment]
In patch "testcases/kernel-bootrr.yaml: Import fix from CI" the wrong sha is provided.
please change the changelog, maybe something like:
testcases/kernel-bootrr: Fix record of lava results

then add some explination...

Link: https://git.linaro.org/ci/job/configs.git/commit/lt-qcom/lava-job-definitions/testplan/kernel-bootrr.yaml?id=80655e6d96bbdeaba7a20a92027e7a3f3f30eb89
Signed-off-by: Aníbal Limón <anibal.limon@linaro.org>

---------------------------------------------------------
In patch "testcases: Import lt-qcom kernel-{desktop, multimedia, smoke} cases"
Please change the changelog to something like this:
testcases: add new tests kernel-{desktop, multimedia, smoke}

<<Describe the individual tests>>

These tests are based on the lt-qcom tests kernel-{desktop, multimedia, smoke}
cases from
https://git.linaro.org/ci/job/configs.git/tree/lt-qcom/lava-job-definitions/testplan
Rev: b7e172fcbcd2599110a76be4cfbec02a2f676226

Signed-off-by: Aníbal Limón <anibal.limon@linaro.org>

---------------------------------------------------------
Patch "testplans: Add lt-qcom-kernel", can you use testplan/lt-qcom/

---------------------------------------------------------
Patch "testcases/kernel-{desktop, multimedia}: Add defaults for couple of vars" should be squashed into patch "testcases: Import lt-qcom kernel-{desktop, multimedia, smoke} cases"

**Comment by alimon** [comment]
> In patch "testcases/kernel-bootrr.yaml: Import fix from CI" the wrong sha is provided.
> please change the changelog, maybe something like:
> testcases/kernel-bootrr: Fix record of lava results
> 
> then add some explination...
> 
> Link: https://git.linaro.org/ci/job/configs.git/commit/lt-qcom/lava-job-definitions/testplan/kernel-bootrr.yaml?id=80655e6d96bbdeaba7a20a92027e7a3f3f30eb89
> Signed-off-by: Aníbal Limón [anibal.limon@linaro.org](mailto:anibal.limon@linaro.org)
> 
> In patch "testcases: Import lt-qcom kernel-{desktop, multimedia, smoke} cases"
> Please change the changelog to something like this:
> testcases: add new tests kernel-{desktop, multimedia, smoke}
> 
> <>
> 
> These tests are based on the lt-qcom tests kernel-{desktop, multimedia, smoke}
> cases from
> https://git.linaro.org/ci/job/configs.git/tree/lt-qcom/lava-job-definitions/testplan
> Rev: b7e172fcbcd2599110a76be4cfbec02a2f676226
> 
> Signed-off-by: Aníbal Limón [anibal.limon@linaro.org](mailto:anibal.limon@linaro.org)
> 
> Patch "testplans: Add lt-qcom-kernel", can you use testplan/lt-qcom/
> 
> Patch "testcases/kernel-{desktop, multimedia}: Add defaults for couple of vars" should be squashed into patch "testcases: Import lt-qcom kernel-{desktop, multimedia, smoke} cases"

@roxell I did the changes you suggest, I would like to have lt-qcom-kernel since we have different testplans one for the kernel and other one for the released images.

**Comment by alimon** [comment]
@roxell I added LXC support to the include/fastboot.jinja2 template in order to migrate our jobs to this repo, my plan is  to move to docker after integrate using LXC, We have a blocker about create a Docker image with tools/keys to access QCOM remote lab and docker support for NFS that is in progress [1].

[1] https://git.lavasoftware.org/lava/lava/-/merge_requests/1401

File: include/fastboot.jinja2 [hunk 1]
```diff
@@ -3,7 +3,6 @@
 {% set use_context = true %}
 
 {% set DOCKER_IMAGE = DOCKER_IMAGE|default("debian:buster") %}
-{% set install_fastboot = install_fastboot|default(true) %}
```
Comment (roxell) [inline include/fastboot.jinja2:6]: This belong to its own commit with the other 'install_fastboot' that gets removed.

**Comment by roxell** [comment]
Can you drop patch: "devices/dragonboard-*: Add venus encoder/decoder video devices." ?
You already have a default in "testcases: Add new lt-qcom kernel-{desktop, multimedia, smoke} cases", can't they be the same?

with the lt-qcom testplans I meant that you can remove the links what isn't needed in for you in testplans/lt-qcom/* and just add the once you need.

**Comment by alimon** [comment]
> Can you drop patch: "devices/dragonboard-*: Add venus encoder/decoder video devices." ?
> You already have a default in "testcases: Add new lt-qcom kernel-{desktop, multimedia, smoke} cases", can't they be the same?
> 
> with the lt-qcom testplans I meant that you can remove the links what isn't needed in for you in testplans/lt-qcom/* and just add the once you need.

Ack.

File: testcases/kernel-desktop.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{% extends "testcases/master/template-master.jinja2" %}
+
+{% set test_timeout = 150 %}
+{% set BOOT_OS_PROMPT = 'root@qemuarm64:~#' %}
```
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-desktop.yaml:4]: @roxell The BOOT_OS_PROMPT in this testcase is different becasue we are using a common ramdisk (qemuarm64) across platforms that's the reason to set at this level.

File: testcases/kernel-desktop.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+{% set test_name = test_name | default("desktop") %}
+
+{% block test_target %}
+  {{ super() }}
+    - repository: https://github.com/Linaro/test-definitions.git
```
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-desktop.yaml:12]: > Please use:
> 
> * repository: {{ TEST_DEFINITIONS_REPOSITORY }}
> 
> the same goes for the other places too.

Ack.

File: testcases/kernel-multimedia.yaml [hunk 4]
```diff
@@ -5,6 +5,8 @@
 
 {% set test_name = test_name | default("multimedia") %}
 
+{% set video_devices = video_devices | default({"/dev/video0": "v4l2-video0"}) %}
```
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-multimedia.yaml:8]: It is a bit complicated, i'm using internal video_devices and VIDEO_DEVICES inside devices to allow override from external variable, if not always be set by video_devices in this case. I can remove this default in kernel-multimedia and use only VIDEO_DEVICES.

File: testcases/kernel-smoke.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+{% set test_name = test_name | default("smoke") %}
+
+{% block test_target %}
+  {{ super() }}
+    - repository: https://git.linaro.org/qa/test-definitions.git
+      from: git
+      path: automated/linux/smoke/smoke.yaml
+      params:
+              TESTS: {{SMOKE_TESTS}}
```
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-smoke.yaml:14]: Ack.

File: testcases/kernel-smoke.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      name: linux-usb-smoke
+    - repository: https://git.linaro.org/qa/test-definitions.git
+      from: git
+      path: automated/linux/meminfo/meminfo.yaml
+      name: meminfo
+    - repository: https://git.linaro.org/qa/test-definitions.git
+      from: git
+      path: automated/linux/hci-smoke/hci-smoke.yaml
+      params:
+              BOOT: auto
```
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-smoke.yaml:28]: Ack,

File: devices/qcs404-evb-1k [hunk 7]
```diff
@@ -4,7 +4,8 @@
 {% set DEPLOY_OS = DEPLOY_OS|default("oe") %}
 {% set ROOTFS_URL_COMP = ROOTFS_URL_COMP|default("gz") %}
 
-{% set install_fastboot = false %}
+{% set lxc_install_fastboot = false %}
+{% set lxc_install_packages = false %}
```
[the remark this answers is not in the document]
Comment (alimon) [reply devices/qcs404-evb-1k:8]: This devices/qcs404-evb-4k and devices/sdm845-mtp are remote and the host has the tools already installed so not need to install again that's the reason for this variable.

File: include/fastboot.jinja2 [hunk 8]
```diff
@@ -141,6 +140,11 @@ reboot_to_fastboot: {{ reboot_to_fastboot }}
 {% endif %}
 {% include "include/boot_target/boot_os_prompt.jinja2" %}
 {% include "include/boot_target/timeout_and_method.jinja2" %}
+{% if fastboot_commands is defined %}
+    commands:
+{% for fcmd in fastboot_commands %}
+       - {{ fcmd }}
+{% endfor %}
```
[the remark this answers is not in the document]
Comment (alimon) [reply include/fastboot.jinja2:147]: I'm using 

{% if fastboot_commands is defined %}
...
{% for fcmd in fastboot_commands %}
       - {{ fcmd }}
{% endfor %}
...
{% endif %}

File: include/fastboot.jinja2 [hunk 9]
```diff
@@ -141,6 +140,11 @@ reboot_to_fastboot: {{ reboot_to_fastboot }}
 {% endif %}
 {% include "include/boot_target/boot_os_prompt.jinja2" %}
 {% include "include/boot_target/timeout_and_method.jinja2" %}
+{% if fastboot_commands is defined %}
+    commands:
+{% for fcmd in fastboot_commands %}
+       - {{ fcmd }}
```
[the remark this answers is not in the document]
Comment (alimon) [reply include/fastboot.jinja2:146]: Ack.

File: projects/lt-qcom/variables-db845c-rootfs.ini [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+LXC_BOOT_FILE=boot-rootfs-linux-integration-v5.11-rc5-319-g2d623fc6ac5c-738-sdm845-db845c.img
+PTABLE_URL=https://images.validation.linaro.org/snapshots.linaro.org/96boards/dragonboard845c/linaro/rescue/28/dragonboard-845c-bootloader-ufs-linux-28/gpt_both0.bin
+LXC_PTABLE_FILE=gpt_both0.bin
+ROOTFS_URL=https://snapshots.linaro.org/member-builds/qcomlt/linux-integration/sdm845-db845c/738/rpb-console-image-test-qemuarm64-20210124073924-672.rootfs.img.gz
+ROOTFS_URL_COMP=gz
+LXC_ROOTFS_FILE=rpb-console-image-test-qemuarm64-20210124073924-672.rootfs.img
+DEPLOY_OS=oe
+OS_INFO=kernel
+BOOT_OS_PROMPT=
+fastboot_commands=oem select-display-panel none, reboot
```
[the remark this answers is not in the document]
Comment (alimon) [reply projects/lt-qcom/variables-db845c-rootfs.ini:17]: Ok will change fastboot_commands to FASTBOOT_COMMANDS.

File: include/fastboot.jinja2 (hunk 1 above)
Comment (alimon) [reply include/fastboot.jinja2:6]: Ack will squash.

File: testcases/kernel-desktop.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-desktop.yaml:4]: @roxell Ack I will drop BOOT_OS_PROMPT from testcases to be set outside from environment or variable file.

File: testcases/kernel-multimedia.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (alimon) [reply testcases/kernel-multimedia.yaml:8]: @roxell Ack I will remove video_devices and leave set by device or from env/file.

**Review by nareshkamboju — APPROVED** [review]
I have boot tested on hikey, db410c, x15 and x86 and all boot PASS.
Considering it as review Approved.

**Comment by nareshkamboju** [comment]
Here is list of boot tested on db410c, hikey, Juno-r2, x15 and x86_64
and boot PASS on these devices.
https://lkft.validation.linaro.org/scheduler/job/2262298
https://lkft.validation.linaro.org/scheduler/job/2261763
https://lkft.validation.linaro.org/scheduler/job/2261759
https://lkft.validation.linaro.org/scheduler/job/2261175
https://lkft.validation.linaro.org/scheduler/job/2257188
